src/components/Request/Pay/Views/Initial.view.tsx

```diff
@@ -38,36 +44,93 @@ export const InitialView = ({
             setErrorState({ showError: false, errorMessage: '' })
             if (!unsignedTx) return
             await assertValues({ tokenValue: requestLinkData.tokenAmount })
-
             setLoadingState('Sign in wallet')
+            if (selectedChainID === requestLinkData.chainId && selectedTokenAddress === requestLinkData.tokenAddress) {
+                const hash = await sendTransactions({
+                    preparedDepositTxs: { unsignedTxs: [unsignedTx] },
+                    feeOptions: undefined,
+                })
 
-            const hash = await sendTransactions({
-                preparedDepositTxs: { unsignedTxs: [unsignedTx] },
-                feeOptions: undefined,
-            })
+                setLoadingState('Executing transaction')
 
-            setLoadingState('Executing transaction')
+                await peanut.submitRequestLinkFulfillment({
+                    chainId: requestLinkData.chainId,
+                    hash: hash ?? '',
+                    payerAddress: address ?? '',
+                    link: requestLinkData.link,
+                    apiUrl: '/api/proxy/patch/',
+                })
 
-            await peanut.submitRequestLinkFulfillment({
-                chainId: requestLinkData.chainId,
-                hash: hash ?? '',
-                payerAddress: address ?? '',
-                link: requestLinkData.link,
-                apiUrl: '/api/proxy/patch/',
-            })
+                const currentDate = new Date().toISOString()
+                utils.saveRequestLinkFulfillmentToLocalStorage({
+                    details: {
+                        ...requestLinkData,
+                        destinationChainFulfillmentHash: hash ?? '',
+                        createdAt: currentDate,
+                    },
+                    link: requestLinkData.link,
+                })
 
-            const currentDate = new Date().toISOString()
-            utils.saveRequestLinkFulfillmentToLocalStorage({
-                details: {
-                    ...requestLinkData,
-                    destinationChainFulfillmentHash: hash ?? '',
-                    createdAt: currentDate,
-                },
-                link: requestLinkData.link,
-            })
+                setTransactionHash(hash ?? '')
+                onNext()
+            } else {
+                // TODO: Check this
+                // const squid = new Squid({ integratorId: 'squid-sdk' })
```

Comment:
fyi: integratorId is sensitive info

btw we also have `get-squid-route.ts` route in api. not sure if it helps here